### ISSUE TITLE
Add class weights for optimizing the loss of tfa.text.crf model

### ISSUE THREAD

**Issue by snailcoder** [tensorflow/addons#817]
**Describe the feature and the current behavior/state.**
It is necessary to add class weights when optimizing the loss of classification model on imbalance data. For example, when use tf.nn.softmax_cross_entropy_with_logits to calculate the model's loss, we can obtain a cross entropy tensor of which the shape is the same as labels so we can multiply class weights with the cross entropy to get a weighted loss. Similarly, when use tfa.text.crf, we should also get the weighted loss.

**Relevant information**
- Are you willing to contribute it (yes/no): no
- Are you willing to maintain it going forward? (yes/no): no
- Is there a relevant academic paper? (if so, where): no
- Is there already an implementation in another framework? (if so, where): no
- Was it part of tf.contrib? (if so, where): no

**Which API type would this fall under (layer, metric, optimizer, etc.)**
metric
**Who will benefit with this feature?** 
All tfa.text.crf users
**Any other info.**

**Comment by Squadrick** [comment]
Hello, `tfa.text.crf` is only for base-level ops. We have a PR for adding CRF layer in #377 

Once we have the basic version of CRF layer merged, we'll get to adding in class weights to the loss.

**Comment by dhakrasp** [comment]
Can anyone guide me as to how I could go about modifying the current implementation of CRF loss so as to include class weights?

**Comment by jaspersjsun** [comment]
@bhack Thanks for cc me. I'm not sure whether it is necessary to add it to the official losses. Do we have further papers, materials or discussions to make this more convincing?  

Since only changing the emission scores (`tfa.text.crf.crf_unary_score`) could be problematic, [Gael de Lannoy et al.](https://perso.uclouvain.be/michel.verleysen/papers/ieeetbe12gdl.pdf) proposed an implementation for weighted CRF. Maybe try to follow it and create a custom loss function with existing ops in `tfa.text.crf` could be an option.

**Comment by eraldoluis** [comment]
@jaspersjsun when you say that changing the emission scores could be problematic, you mean theoretically or in terms of the code base?

I'm studying this issue with the intent of implementing some approach (first in AllenNLP, then I could do it in TF as well). I think Lannoy et al.'s approach is theoretically flawed. I mean, I believe it can work in practice, but their theoretical motivation is not correct. If this is the case, we could do it with much simpler approach (like weighted emission scores).

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
